### PR TITLE
[Enhancement] support iso week function (#20511)

### PR DESCRIPTION
**Pull request by qmengss**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #22773

## Checklist:

- [X] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by imay** [comment]
Thank you for your contribution. Please add documents about this newly added function.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by qmengss** [comment]
> about

doc had add.

File: docs/sql-reference/sql-functions/date-time-functions/week_iso.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# week_iso
```
Comment (imay) [inline docs/sql-reference/sql-functions/date-time-functions/week_iso.md:1]: Should add this function in docs/toc.md

**Review by imay — COMMENTED** [review]
Also you can try writing sql-test which is more convenient. It is in test/ directory

File: docs/sql-reference/sql-functions/date-time-functions/week_iso.md (hunk 1 above)
Comment (qmengss) [reply docs/sql-reference/sql-functions/date-time-functions/week_iso.md:1]: done

**Comment by qmengss** [comment]
> 

let me have a look.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by qmengss** [comment]
> 
test-sql has added.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — DISMISSED** [review]

File: test/sql/test_time_fn/T/test_time_fn [hunk 2]
```diff
@@ -181,3 +181,9 @@ select d3, dow_2, previous_day(d3, dow_2) from previous_dow order by d3;
 select d3, dow_3, previous_day(d3, dow_3) from previous_dow order by d3;
 select d3, dow_full, previous_day(d3, dow_full) from previous_dow order by d3;
 select d1, dow_2, previous_day(d1, dow_2) from previous_day_test a join previous_dow b on a.d1 = b.d3 order by d1;
+
+
+-- name: test_week_iso
+select week_iso("2023-01-01");
+select week_iso("2023-01-02");
+select week_iso("2023-01-03");
```
Comment (imay) [inline test/sql/test_time_fn/T/test_time_fn:189]: can you add a case to test `week_iso(null)`

Comment (qmengss) [reply test/sql/test_time_fn/T/test_time_fn:189]: Do you mean week_iso("")？

Comment (qmengss) [reply test/sql/test_time_fn/T/test_time_fn:189]: > can you add a case to test `week_iso(null)`
week_iso(null) and week_iso("") has added to the test-sql.

**Review by imay — APPROVED** [review]

**Comment by LiShuMing** [comment]
Well done~ LGTM.

**Comment by Youngwb** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**